src/ORM/Association.php

```diff
@@ -751,6 +751,20 @@ public function deleteAll($conditions)
     }
 
     /**
+     * Returns true if the eager loading process will require a set of the owning table's
+     * binding keys in order to use them as a filter in the finder query.
+     *
+     * @param array $options The options containing the strategy to be used.
+     * @return bool true if a list of keys will be required
+     */
+    public function requiresKeys(array $options = [])
+    {
+        $strategy = isset($options['strategy']) ? $options['strategy'] : $this->strategy();
```

Comment:
It seems to also act as a setter and thus the docblock above should educate about that.